usecases/Concrete/knowledgeGraph/emodul/emodul_metadata_extraction.py

```diff
@@ -96,42 +97,73 @@ def eModul_metadata(filePath, fileName ):
 
 # print(data)
 
-metadata = []
-for folder in listDataFolders:
-    path = os.path.join(dataPath,folder)
-    try:
-        metadata.append(eModul_metadata(path, 'specimen.dat'))
-    except:
-        print('something wrong')
-        print(folder)
+
 
 def column_data(listName, index, firstKey, secondKey):
     data = []
     for i in listName:
         data.append(i[index][firstKey][secondKey])
     return data
 
-experimentNameIndex = 0
-dataTypeIndex = 1
-operatorIndex = 2
-dataCollectionIndex = 3
-
-dataFrame = pd.DataFrame({
-    'experiment raw name': [i[experimentNameIndex]['experimentName'] for i in metadata],
-    'data type': [i[dataTypeIndex]['data type'] for i in metadata],
-    'operator time': column_data(metadata, operatorIndex, 'Bediener Information', 'Zeit'),
-    'operator timestamp': column_data(metadata, operatorIndex, 'Bediener Information', 'Zeitpunkt'),
-    'operator date': column_data(metadata, operatorIndex, 'Bediener Information', 'Datum'),
-    'tester': column_data(metadata, operatorIndex, 'Bediener Information', 'Prfer'),
-    'sample name': column_data(metadata, operatorIndex, 'Bediener Information', 'Probenbezeichnung'),
-    'remark': column_data(metadata, operatorIndex, 'Bediener Information', 'Bemerkungen'),
-    'weight': column_data(metadata, operatorIndex, 'Bediener Information', 'Masse'),
-    'diameter': column_data(metadata, operatorIndex, 'Bediener Information', 'Durchmesser'),
-    'length': column_data(metadata, operatorIndex, 'Bediener Information', 'Lnge'),
-    'data collection time': column_data(metadata, dataCollectionIndex, 'Datenerfassung', 'Zeit'),
-    'data collection timestamp': column_data(metadata, dataCollectionIndex, 'Datenerfassung', 'Zeitpunkt')
-})
-
-dataFrame.to_csv(os.path.join(baseDir0,'E-modul-processed-data/emodul_metadata.csv'), index = False)
-
-print(dataFrame.head())
\ No newline at end of file
+def generate_metadata_yaml_file_from_emodul_BAM(locationOfRawData, locationOfMetaData):
```

Comment:
I think the return function should rather be the dict, that you can then store in a separate file if needed (this way you can test the functionality without intermediate files in pytest)